packages/core/src/managers/stamina-manager.ts

```
@@ -28,53 +24,35 @@ export class StaminaManager {
-    components: ClientComponents,
-  ) {
-    const last_refill_tick = staminaValue.updated_tick;
+  public static getStamina(troops: Troops, currentArmiesTick: number) {
+    const last_refill_tick = troops.stamina.updated_tick;
 
     if (last_refill_tick >= BigInt(currentArmiesTick)) {
-      return structuredClone(staminaValue);
+      return structuredClone(troops.stamina);
     }
 
     const newStamina = this.refill(
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)